metrics/src/metrics.rs

```diff
@@ -460,6 +460,31 @@ pub fn set_panic_hook(program: &'static str) {
     });
 }
 
+lazy_static! {
+    static ref IS_CANARY_VALIDATOR: Arc<RwLock<bool>> = Arc::new(RwLock::new(false));
+}
+
+pub fn set_is_canary_validator(is_canary_validator: bool) {
+    info!("Setting IS_CANARY_VALIDATOR: {}", is_canary_validator);
+    *IS_CANARY_VALIDATOR.write().unwrap() = is_canary_validator;
+}
+
+pub fn canary_assert(assertion: bool, failure_message: &str) {
+    let is_canary_validator = *IS_CANARY_VALIDATOR.read().unwrap();
```

Comment:
Looks like `set_is_canary_validator()` can disappear if a `std::sync::Once` is employed in here to read the env var